### PR TITLE
Add JSON support

### PR DESCRIPTION
**Pull request by vadi2**
# map save - save the map

Use: \*map save \<optional name\>\* will save all of the map. Sample use:

    default map - ie 'map save' will save the map with the default name as the latest one
    map name - ie 'map save map before experiment' will save the map with that name
    custom - ie 'map save custom' will export the map as a file, so you can share it with others

You can also save the map as text (JSON) by using `map save json` or `map save all` (in both normal and JSON). For example:

    default map - ie 'map save json' will save the map with the default name as the latest one, 'map save all' will do it in both.
    map name - ie 'map save json map before experiment' will save the map with that name
    custom - ie 'map save json custom' will export the map as a file, so you can share it with others

# map load - load the map

Use: \*map load \<optional name\>\* will reload the map. Sample use:

    default map - ie 'map load' will load the latest saved map (it may be a named one)
    map name - ie 'map load before experiment' will load the map with that name
    custom - ie 'map load custom' will load a map .dat file as you select it
    as json - 'map load before experiment.json' - appending '.json' will make it load as json

### CODE REVIEW

File: mudlet-mapper.xml [hunk 1]
```diff
@@ -4915,53 +4915,91 @@ end</script>
...
+    savednormal = saveMap(location)
+    savedjson = saveJsonMap(location)
+  elseif format == nil or format == "" then
+    savednormal = saveMap(location)
+  end
+  
+  if savednormal == nil then
+    mmp.echo("Couldn't save the map :(")
+  elseif savednormal == true then
+    mmp.echo(loc ~= "" and "Map saved." or "Saved the default map.")
```
Comment (keneanung) [inline mudlet-mapper.xml:4932]: Shouldn't this be `location ~= `?

File: mudlet-mapper.xml [hunk 2]
```diff
@@ -4915,53 +4915,91 @@ end</script>
...
+  end
+  if savedjson == nil then
+    mmp.echo("Couldn't save the JSON map :(")
+  elseif savedjson == true then
+    mmp.echo(loc ~= "" and "Map saved in JSON." or "Saved the default map in JSON.")
```
Comment (keneanung) [inline mudlet-mapper.xml:4937]: Shouldn't this be `location ~= `?

**Comment by vadi2** [comment]
Fixed.

**Review by keneanung — APPROVED** [review]
Fine with this now. Let's get it out today. But in general outputting the file name(s) including path would be great to find the file again.

**Pull request merged**